app/src/main/java/com/bleyl/recurrence/database/DatabaseHelper.java

```diff
@@ -119,6 +123,13 @@ public void onUpgrade(SQLiteDatabase database, int oldVersion, int newVersion) {
                     + COL_FRIDAY + " TEXT, "
                     + COL_SATURDAY + " TEXT) ");
         }
+
+        if (oldVersion < 4) {
+            database.execSQL("ALTER TABLE " + NOTIFICATION_TABLE + " ADD " + COL_NAG_TIMER + " INTEGER");
+            database.execSQL("ALTER TABLE " + NOTIFICATION_TABLE + " ADD " + COL_ACTIVE_STATE + " BOOLEAN");
+            database.execSQL("UPDATE " + NOTIFICATION_TABLE + " SET " + COL_NAG_TIMER + " = 0;");
+            database.execSQL("UPDATE " + NOTIFICATION_TABLE + " SET " + COL_ACTIVE_STATE + " = FALSE;");
```

Comment:
It seems it worked, I could upgrade from database V3 without crash and keeping set notifications. Thanks a lot for your help.